vmngclient/endpoints/configuration/software_actions.py

```diff
@@ -0,0 +1,187 @@
+        default=None, serialization_alias="networkFunctionType", validation_alias="networkFunctionType"
+    )
+    platform_family: Optional[List[str]] = Field(
+        default=None, serialization_alias="platformFamily", validation_alias="platformFamily"
+    )
+    rid: Optional[int] = Field(default=None, serialization_alias="@rid", validation_alias="@rid")
+    smu_compatible_version: Optional[str] = Field(
+        default=None, serialization_alias="smuCompatibleVersion", validation_alias="smuCompatibleVersion"
+    )
+    system_properties_xml: Optional[str] = Field(
+        default=None, serialization_alias="systemPropertiesXml", validation_alias="systemPropertiesXml"
+    )
+    updated_on: Optional[int] = Field(default=None, serialization_alias="updatedOn", validation_alias="updatedOn")
+    vendor: Optional[str] = Field(default=None)
+    version_id: Optional[str] = Field(default=None, serialization_alias="versionId", validation_alias="versionId")
+    version_id_map: Optional[str] = Field(
+        default=None, serialization_alias="versionIdMap", validation_alias="versionIdMap"
+    )
+    version_name: Optional[str] = Field(default=None, serialization_alias="versionName", validation_alias="versionName")
+    version_type: Optional[str] = Field(default=None, serialization_alias="versionType", validation_alias="versionType")
+    version_type_name: Optional[str] = Field(
+        default=None, serialization_alias="versionTypeName", validation_alias="versionTypeName"
+    )
+    version_url: Optional[str] = Field(default=None, serialization_alias="versionURL", validation_alias="versionURL")
+    vnf_properties_json: Optional[str] = Field(
+        default=None, serialization_alias="vnfPropertiesJson", validation_alias="vnfPropertiesJson"
+    )
+
+
+class ConfigurationSoftwareActions(APIEndpoints):
+    @get("/device/action/remote-server", "data")
+    def get_list_of_remote_servers(self) -> DataSequence[RemoteServerInfo]:
+        ...
+
+    @post("/device/action/remote-server")
+    def add_new_remote_server(self, payload: RemoteServer) -> None:
+        ...
+
+    @get("/device/action/remote-server/{id}", "data")
+    def get_remote_server(self, id: UUID) -> RemoteServerInfo:
+        ...
+
+    @put("/device/action/remote-server/{id}")
+    def update_remote_server(self, id: UUID) -> None:
+        ...
+
+    @delete("/device/action/remote-server/{id}")
+    def remove_remote_server(self, id: UUID) -> None:
+        ...
+
+    @get("/device/action/software", "data")
+    def get_software_images(self) -> DataSequence[SoftwareImageDetails]:
+        ...
+
+    @post("/device/action/software")
+    def upload_software_from_remote_server(self, payload: SoftwareRemoteServer) -> None:
+        ...
+
+    @delete("/device/action/software/{id}")
+    def delete_software_from_software_repository(self, id: UUID) -> None:
+        ...
+
+    @get("/device/action/software/images?imageType={image_type}", "data")
```

Comment:
let's not hard-code query params in endpoint uri, we can use dedicated `params` keyword argument which can be a raw dict or pydantic DataModel. According to spec for this endpoint following query params are applicable:
`imageType` which is an array? (required)
`vnfType`

Reply:
Done